tools/protodoc/protodoc.py

```diff
@@ -514,23 +526,35 @@ def FormatFieldAsDefinitionListItem(outer_type_context, type_context, field):
   Returns:
     RST formatted definition list item.
   """
+  annotations = []
   if field.HasField('oneof_index'):
-    oneof_template = '\nPrecisely one of %s must be set.\n' if type_context.oneof_required[
-        field.oneof_index] else '\nOnly one of %s may be set.\n'
-    oneof_comment = oneof_template % ', '.join(
-        FormatInternalLink(
-            f, FieldCrossRefLabel(outer_type_context.ExtendField(0, f).name))
-        for f in type_context.oneof_fields[field.oneof_index])
+    oneof_context = outer_type_context.ExtendOneof(field.oneof_index,
+                                                   type_context.oneof_names[field.oneof_index])
+    oneof_comment, oneof_comment_annotations = oneof_context.LeadingCommentPathLookup()
+    if NOT_IMPLEMENTED_HIDE_ANNOTATION in oneof_comment_annotations:
+      return ''
+
+    if len(type_context.oneof_fields[field.oneof_index]) == 1 and type_context.oneof_required[
```

Comment:
Can you add a comment explaining this case?

Reply:
Done